Makefile

```diff
@@ -16,7 +16,12 @@ maintainer-clean : clean
 	$(RM) $(DTBS)
 
 %.dtbo : %.dts
-	dtc -Wno-avoid_default_addr_size -Wno-avoid_unnecessary_addr_size \
+ifndef KERNEL_PATH
+	$(error KERNEL_PATH is not set)
```

Comment:
You don't need an ifdef for this. Its better to conditionally assign (this won't interfere with `make all` since the error doesn't get reported unless something tries to expand KERNEL_PATH):

KERNEL_PATH ?= $(error KERNEL_PATH is not set)

However...

C pre-processing looks sensible but we don't have a single kernel tree to point KERNEL_PATH at. I know dt-update is currently only populated with examples from Dragonboard family... but we don't want things to stay that way.

I think its relatively easily solved by an approach based on target specific variables such as:

DB410C_KERNEL_PATH ?= $(error DB410C_KERNEL_PATH is not set)
DB410C_DTBS = $(wildcard scripts/db410/overlays/*dts
db410c_dtbs : KERNEL_PATH = $(DB410_KERNEL_PATH)

HIKEY620_KERNEL_PATH ?= $(error HIKEY620_KERNEL_PATH is not set)
HIKEY620_DTBS = $(wildcard scripts/hikey620/overlays/*dts
hikey620_dtbs : KERNEL_PATH = $(HIKEY620_KERNEL_PATH)

dtbs : db410c_dtbs hikey620_dtbs

I don't mind if both DB410C and DB820C share a DRAGONBOARD_KERNEL_PATH (since you have a unified kernel) but I'd like to leave space for other boards to be added.

Reply:
Hi Daniel,

Thank you for review. I'll create a new pull request with these suggestions implemented.